pkg/trace/event/processor.go

```diff
@@ -83,14 +76,19 @@ func (p *Processor) Process(root *pb.Span, t pb.Trace) (events []*pb.Span, numEx
 		sampler.SetClientRate(span, clientSampleRate)
 		sampler.SetPreSampleRate(span, preSampleRate)
 		sampler.SetEventExtractionRate(span, extractionRate)
-		if hasPriority {
-			sampler.SetSamplingPriority(span, priority)
+		sampler.SetAnalyzedSpan(span)
+		if t.DroppedTrace {
+			events = append(events, span)
 		}
+		numEvents++
+	}
 
-		events = append(events, span)
+	if t.DroppedTrace {
+		// we are not keeping anything out of this trace, except the events
+		t.Spans = events
 	}
 
```

Comment:
And this one. I tried to explain the logic here: https://github.com/DataDog/dd-trace-go/wiki/Style-guidelines#general-guidelines at point 3. I realise I may be annoying here, but it's hard to put these rules into a linter.